File: 2000.reverse-prefix-of-word.cpp
/*
 * @lc app=leetcode id=2000 lang=cpp
 *
 * [2000] Reverse Prefix of Word
 */
#include <string>
using namespace std;
// @lc code=start
class Solution
{
public:
    string reversePrefix(string word, char ch)
    {
        int pos = word.find(ch);
        if (pos != -1)
        {
            reverse(word.begin(), word.begin() + pos + 1);
        }
        return word;
    }
};
// @lc code=end

File: py/2000.reverse-prefix-of-word.py
#
# @lc app=leetcode id=2000 lang=python3
#
# [2000] Reverse Prefix of Word
#

# @lc code=start
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        pos = word.find(ch)
        if pos != -1:
            word = word[:pos+1][::-1]+word[pos+1:]
        return word
# @lc code=end
